test/e2e/integration_test.go

```diff
@@ -9,6 +9,17 @@ import (
 	"sigs.k8s.io/controller-runtime/pkg/client"
 )
 
+// Integration tests in capm3 focus on validating the seamless integration between different components of the capm3 project,
+// including capm3, IPAM, CAPI, BMO, and Ironic. These tests ensure that these components work together cohesively to provision a
+// workload cluster and perform pivoting operations between the bootstrap cluster and the target cluster.
+// The primary goal is to detect any compatibility issues or conflicts that may arise during integration.
+
+// By executing the integration tests, capm3 verifies that:
+
+// - The capm3 controller effectively interacts with the IPAM, CAPI, BMO, and Ironic components.
+// - The provisioning of a workload cluster proceeds smoothly, creating bmhs, inspecting and provision them.
+// - The pivoting functionality enables the seamless transition of resources and control components from the bootstrap cluster to the target cluster and vice versa.
+// - Deprovision then cluster and wait for the bmh to get deprovisoned.
```

Comment:
Neither does this. Maybe something like "Deprovisioning the cluster and bmhs happens smoothly."

Reply:
done